annorxiver_modules/journal_rec_helper.py

```diff
@@ -0,0 +1,197 @@
+    All the classifer does is randomly pick journals for recommendation.
+    
+    arguments:
+        model - dummy classifier
+        training_data - the data to train the classifier
+        validation_data - the data to evaluate the classifier
+        kwargs - extra keywords to pass if desired
+    """
+    
+    top_X = kwargs.get("top_predictions", 10)
+    random_states = kwargs.get("dummy_seed", [100,200,300,400,500,600,700,800,900,1000])
+    
+    X_train = (
+        training_data
+        .drop("journal", axis=1)
+        .values
+        .astype('float32')
+    )
+    
+    Y_train = (
+        training_data
+        .journal
+        .values
+    )
+    
+    X_val = (
+        validation_data
+        .drop("journal", axis=1)
+        .values
+        .astype('float32')
+    )
+    Y_val = (
+        validation_data
+        .journal
+        .values
+    )
+    
+    predictions = []
+    for i, seed in zip(range(top_X), random_states):
+        model.random_state = seed
+        model.fit(X_train, Y_train)
+        predictions.append(model.predict(X_val))
+
+    return np.stack(predictions).transpose(), Y_val
+
+
+def knn_evaluate(model, training_data, validation_data, **kwargs):
+    """
+    This function is used to evaluate the knearestneighbors classifier 
+    on an individual paper by paper basis.
+    
+    arguments:
+        model - knearestneighbor classifier
+        training_data - the data to train the classifier
+        validation_data - the data to evaluate the classifier
+        kwargs - extra keywords to pass if desired
+    """
+    
+    X_train = (
+        training_data
+        .drop("journal", axis=1)
+        .values
+        .astype('float32')
+    )
+    
+    Y_train = (
+        training_data
+        .journal
+        .values
+    )
+    
+    X_val = (
+        validation_data
+        .drop("journal", axis=1)
+        .values
+        .astype('float32')
+    )
+    
+    Y_val = (
+        validation_data
+        .journal
+        .values
+    )
+    
+    model.fit(X_train, Y_train)
+    distance, neighbors = model.kneighbors(X_val)
+    
+    predictions = [
+        Y_train[neighbor_predict]
+        for neighbor_predict in neighbors 
+    ]
+
+    return np.stack(predictions), Y_val
+
+def knn_centroid_evaluate(model, training_data, validation_data, **kwargs):
+    """
+    This function is used to evaluate the knearestneighbors classifier 
+    by calculating journal centroids for training.
+    
+    arguments:
+        model - knearestneighbor classifier
+        training_data - the data to train the classifier
+        validation_data - the data to evaluate the classifier
+        kwargs - extra keywords to pass if desired
+    """
+    
+    train_centroid_df = (
+        training_data
+        .groupby("journal")
+        .agg("mean")
+        .reset_index()
+    )
+            
+    X_train_centroid = (
+        train_centroid_df
+        .drop("journal", axis=1)
```

Comment:
>I may be wrong because I'm not sure exactly what the columns of this dataframe are, but is this logic behind this selection "Pick the column that isn't 'journal'"?

The dataframe contains the column that specifies the journal each article belongs and the document vectors of 300 dimensions. 

>If so, wouldn't it be more explicit to select by column like you do in Y_train_centroid?

To save time it is easier to just drop the journal column rather than select all the features.